docs: document the components fixture in the e2e testing docs (#2816)
grafana/plugin-tools: docusaurus/docs/e2e-test-a-plugin/selecting-ui-elements.md

Here are some examples demonstrating how to interact with a few UI components that are common in plugins. The `InlineField` and `Field` component can be used interchangeably.

> For many common Grafana UI components, such as select, switch, radio group, and color picker, `@grafana/plugin-e2e` provides ready-made component models through the [`components` fixture](./use-the-api.md#components). Prefer the `components` fixture whenever it covers the component you're testing - it handles UI differences between Grafana versions and actionability quirks for you.

Reviewer: ```suggestion
For many common Grafana UI components, such as `select`, `switch`, `radio group`, or `color picker`, `@grafana/plugin-e2e` provides ready-made component models through the [`components` fixture](./use-the-api.md#components). When it covers the component you're testing, use the `components` fixture, since it handles UI differences between Grafana versions and other elements and the component you're targeting.
```

Reviewer: @mckn - I guess this is what you mean by "actionability quirks"?

Author: Not really, it handles differences between versions and also applies fixes where some of the components is hard to interact with at certain places in Grafana. It could be that other UI elements makes it hard to, via tests, interact with the component you are targeting.

Reviewer: Gotcha, updated the suggestion, LMK if it makes better sense.

You can use the `InlineSwitch` component to interact with the UI.

```tsx title="UI component"
<div data-testid="config-tls-field">
  <InlineField label="TLS Enabled">
    <InlineSwitch label="TLS Enabled" value={value} onChange={handleOnChange} />
  </InlineField>
</div>
```

> Use the `switch` model from the [`components` fixture](./use-the-api.md#components) to interact with the component. It resolves the switch element across Grafana versions, and its `check` and `uncheck` methods are idempotent - they only click the switch when the state needs to change, which avoids races with React's re-render cycle.

Reviewer: ```suggestion
Use the `switch` model from the [`components` fixture](./use-the-api.md#components) to interact with the component. It resolves the switch element across Grafana versions, and its `check` and `uncheck` methods are idempotent, so they only click the switch when the state needs to change, which avoids races with React's re-render cycle.
```
